From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Wed, 20 Nov 2024 11:28:34 +0100
Subject: tm: remove destroy of internal structures on mod-destroy callback

- it is safer and faster to be removed at once by core or OS when application
  context is destroyed
---
 src/modules/tm/t_funcs.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/src/modules/tm/t_funcs.c b/src/modules/tm/t_funcs.c
index 5515a4a..92a7e79 100644
--- a/src/modules/tm/t_funcs.c
+++ b/src/modules/tm/t_funcs.c
@@ -87,23 +87,6 @@ int send_pr_buffer(struct retr_buf *rb, void *buf, int len
 
 void tm_shutdown()
 {
-
-	LM_DBG("start\n");
-
-#ifdef USE_DNS_FAILOVER
-	if(failover_reply_codes)
-		shm_free(failover_reply_codes);
-	if(failover_reply_codes_cnt)
-		shm_free(failover_reply_codes_cnt);
-#endif
-	/* destroy the hash table */
-	LM_DBG("emptying hash table\n");
-	free_hash_table();
-	LM_DBG("removing semaphores\n");
-	lock_cleanup();
-	LM_DBG("destroying tmcb lists\n");
-	destroy_tmcb_lists();
-	free_tm_stats();
 	LM_DBG("done\n");
 }
 
